// @mongosh/cli-repl/lib/build-info.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.buildInfo = exports.baseBuildInfo = void 0;
const os_1 = __importDefault(require("os"));
const service_provider_server_1 = require("@mongosh/service-provider-server");
function getSystemArch() {
    return process.platform === 'darwin'
        ? os_1.default.cpus().some((cpu) => {
            return /Apple/.test(cpu.model);
        })
            ? 'arm64'
            : 'x64'
        : process.arch;
}
function baseBuildInfo() {
    const runtimeData = {
        nodeVersion: process.version,
        opensslVersion: process.versions.openssl,
        sharedOpenssl: !!process.config.variables.node_shared_openssl,
        runtimeArch: getSystemArch(),
        runtimePlatform: process.platform,
    };
    try {
        return {
            ...require('./build-info.json'),
            ...runtimeData,
        };
    }
    catch (_a) {
        const { version } = require('../package.json');
        return {
            version,
            distributionKind: 'unpackaged',
            buildArch: process.arch,
            buildPlatform: process.platform,
            buildTarget: 'unknown',
            buildTime: null,
            gitVersion: null,
            ...runtimeData,
        };
    }
}
exports.baseBuildInfo = baseBuildInfo;
async function buildInfo({ withSegmentApiKey, } = {}) {
    const dependencyVersionInfo = {
        ...service_provider_server_1.CliServiceProvider.getVersionInformation(),
    };
    const buildInfo = { ...baseBuildInfo(), deps: { ...dependencyVersionInfo } };
    if (!withSegmentApiKey) {
        delete buildInfo.segmentApiKey;
    }
    return buildInfo;
}
exports.buildInfo = buildInfo;
